server/lib/session/session-api.ts

```diff
@@ -96,10 +98,11 @@ export class SessionApi {
         // TODO(tec27): Make this required in future versions (cur v8.0.2). This is just to allow
         // old clients to log in so it triggers auto-update
         clientIds: joiClientIdentifiers(),
+        language: Joi.valid(...ALL_TRANSLATION_LANGUAGES),
```

Comment:
As stated in the user type, I don't think it makes sense to validate this way. Would be better to accept any BCP-47 tag probably (but sort of depends what i18next's thing is doing). It's very possible you should just be sending whatever value i18next looks at to do "detection" instead of using their resolved value for this, as I assume the purpose of their stuff is "given a known browser language, pick from the languages that are enabled" rather than "what is the browser language"

Reply:
Also I can't leave a comment on it, but imo you should be adding a query parameter to `getCurrentSession` that includes a language code, and migrate existing accounts if they don't have a language set in that